### PR TITLE
feature-261: Role Delete

### PR DESCRIPTION
**Pull request by atye**
# Description

Adds `karavictl role delete` functionality to the role-service, updates some logging, and fixes and issue in `role create` where multiple roles are not handled.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/261 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Unit tests and manual tests with helm deployment verifying that the relevant configMap has correct data.

```
go test -count=1 -cover -race -timeout 30s -short ./...
?       karavi-authorization/cmd/karavictl      [no test files]
ok      karavi-authorization/cmd/karavictl/cmd  2.186s  coverage: 61.3% of statements
ok      karavi-authorization/cmd/proxy-server   0.060s  coverage: 4.0% of statements
?       karavi-authorization/cmd/role-service   [no test files]
?       karavi-authorization/cmd/sidecar-proxy  [no test files]
ok      karavi-authorization/cmd/tenant-service 0.039s  coverage: 8.7% of statements
ok      karavi-authorization/deploy     0.139s  coverage: 87.5% of statements
?       karavi-authorization/internal/decision  [no test files]
ok      karavi-authorization/internal/k8s       0.133s  coverage: 56.4% of statements
ok      karavi-authorization/internal/powerflex 9.525s  coverage: 92.6% of statements
ok      karavi-authorization/internal/proxy     9.364s  coverage: 66.1% of statements
ok      karavi-authorization/internal/quota     0.501s  coverage: 90.1% of statements
ok      karavi-authorization/internal/role-service      0.039s  coverage: 76.7% of statements
ok      karavi-authorization/internal/role-service/roles        0.054s  coverage: 93.2% of statements
ok      karavi-authorization/internal/role-service/validate     0.235s  coverage: 79.4% of statements
ok      karavi-authorization/internal/tenantsvc 2.638s  coverage: 82.3% of statements
ok      karavi-authorization/internal/token     0.038s  coverage: 100.0% of statements
ok      karavi-authorization/internal/token/jwx 0.046s  coverage: 71.7% of statements
?       karavi-authorization/internal/types     [no test files]
ok      karavi-authorization/internal/web       0.047s  coverage: 6.2% of statements
?       karavi-authorization/pb [no test files]
```

### CODE REVIEW

File: cmd/karavictl/cmd/role_delete_grpc_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
+// Copyright © 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (atye) [reply cmd/karavictl/cmd/role_delete_grpc_test.go]: Updated.

**Review by tdawe — APPROVED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Review by sharmilarama — DISMISSED** [review]

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Review by shaynafinocchiaro — DISMISSED** [review]

**Review by hoppea2 — DISMISSED** [review]

**Comment by atye** [comment]
run e2e test

**Review by alikdell — APPROVED** [review]

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Review by alikdell — APPROVED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Pull request merged**